### PR TITLE
Merge Gemfiles.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [ ] Have you successfully run `brew tests` with your changes locally?

---

### CODE REVIEW

File: Library/Homebrew/dev-cmd/tests.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def tests
 
       Homebrew.install_gem_setup_path! "bundler"
       unless quiet_system("bundle", "check")
-        system "bundle", "install", "--path", "vendor/bundle"
+        system "bundle", "install"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/tests.rb]: Is this specified somewhere else now?

File: Library/Homebrew/test/Gemfile [hunk 2]
```diff
@@ -1,9 +1,15 @@
 source "https://rubygems.org"
 
-gem "mocha", "~> 1.1"
-gem "minitest", "~> 5.3"
-gem "rake", "~> 10.3"
-gem "parallel_tests", "~> 2.9"
+group :test do
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/Gemfile]: Given these are only used for tests this seems a bit redundant but don't have strong feelings.

File: Library/Homebrew/test/Gemfile [hunk 3]
```diff
@@ -1,9 +1,15 @@
...
+  gem "minitest", "~> 5.9"
+  gem "minitest-reporters"
+  gem "mocha", "~> 1.2"
+  gem "rake", "~> 10.5"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/Gemfile]: Given a bunch of these are being bumped let's just remove all the version pinning here (we get it from `.lock` anyway) and restrict it to the cases where the newest version breaks `brew tests`

File: Library/Homebrew/test/Gemfile [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
+  gem "rspec", "~> 3.5.0"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/Gemfile]: As mentioned before: I'm a very strong 👎 on installing MiniTest and RSpec for the unified `brew tests`. We need to pick one test suite and use that.

File: Library/Homebrew/dev-cmd/tests.rb (hunk 1 above)
Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/tests.rb]: It's the default.

Comment (jawshooah) [inline Library/Homebrew/dev-cmd/tests.rb]: It's not the default, but there is likely a `.bundle/config` in the same directory as the `Gemfile` that specifies this behavior. It is created the first time you run `bundle install` with `--path`.

Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/tests.rb]: Indeed and conversely user global configs will override this unless specified. Better to leave as-is, I think.

File: Library/Homebrew/test/Gemfile (hunk 4 above)
Comment (reitermarkus) [inline Library/Homebrew/test/Gemfile]: I'm very much in favor of the RSpec syntax. Still, we will need to get both working first in order to transition from one to the other. And installing everything in one place is certainly better than it was before, where everything was duplicated also in `cask/vendor/bundle`.

**Comment by MikeMcQuaid** [comment]
> I'm very much in favor of the RSpec syntax. Still, we will need to get both working first in order to transition from one to the other. And installing everything in one place is certainly better than it was before, where everything was duplicated also in cask/vendor/bundle.

I think it would be better to remove the RSpec dependency first before we add it to `brew tests`.

File: Library/Homebrew/test/Gemfile [hunk 5]
```diff
@@ -1,9 +1,13 @@
 source "https://rubygems.org"
 
-gem "mocha", "~> 1.1"
-gem "minitest", "~> 5.3"
-gem "rake", "~> 10.3"
-gem "parallel_tests", "~> 2.9"
+gem "minitest"
```
Comment (jawshooah) [inline Library/Homebrew/test/Gemfile]: We can remove this as well, since `minitest` has been part of the standard library since 1.9.3.

Comment (reitermarkus) [inline Library/Homebrew/test/Gemfile]: I'd keep this, since we aren't using the one from the standard library, but a more up-to-date one.

Comment (MikeMcQuaid) [inline Library/Homebrew/test/Gemfile]: I'm cool with using the newer one.

File: Library/Homebrew/dev-cmd/tests.rb (hunk 1 above)
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/tests.rb]: Still need this changed and then I think we're good.

**Comment by MikeMcQuaid** [comment]
@reitermarkus Any news here? Thanks!

**Comment by MikeMcQuaid** [comment]
@reitermarkus Any news on this one? Thanks!

**Comment by reitermarkus** [comment]
I'll close this, as the gems are now already stored in the same directory for both test suites. So by merging the test suites, cask's Gemfile will eventually be unnecessary anyway.

**Comment by MikeMcQuaid** [comment]
Thanks @reitermarkus!

**Pull request closed without merging**